src/vintch_model/nonlinearity.py

```diff
@@ -0,0 +1,128 @@
+from typing import TypeVar, Generic, Literal
+import einops
+from .backend_config import BackendBase
+import numpy as np
+
+Tensor = TypeVar("Tensor")
+
+
+class TentNonlinearity(Generic[Tensor]):
+    """
+    Tent-based nonlinearity.
+
+    Parameters
+    ----------
+    n_basis_funcs :
+        Number of basis functions to use. Must be a positive integer.
+    backend_instance :
+        Backend to use. Must be an instance of BackendBase.
+    nonlinearity_mode :
+        Type of nonlinearity to initialize weights for. Weights will be fitted to data.
+    """
+
+    def __init__(
+        self,
+        backend_instance: BackendBase,
+        n_basis_funcs: int = 15,
+        nonlinearity_mode: Literal["relu", "quadratic"] = "relu",
+    ):
+        if n_basis_funcs < 1:
+            raise ValueError("basis functions must be a positive integer.")
+
+        if nonlinearity_mode not in ["relu", "quadratic"]:
+            raise ValueError(
+                f"nonlinearity_mode must be 'relu' or 'quadratic', got '{nonlinearity_mode}'"
+            )
+
+        self._n_basis_funcs = n_basis_funcs
+        self._backend = backend_instance
+
+        self._initialize_weights = (
+            self._initialize_relu_weights
+            if nonlinearity_mode == "relu"
+            else self._initialize_quadratic_weights
+        )
+        self._initialize_basis_functions()
+        self._initialize_weights()
```

Comment:
Given that the initialize weights method is never called again, I'd rearrange this. what you have is confusing to me -- I thought the initialize weights methods were being called before `_initialize_basis_functions` and I couldn't figure out how that was possible. 
```suggestion
        self._initialize_basis_functions()
        if nonlinearity_mode == "relu":
            self._initialize_relu_weights()
        elif nonlinearity_mode == "quadratic":
            self._initialize_quadratic_weights()
```